### PR TITLE
Data quality engine enhancements 

### PR DESCRIPTION
**Pull request by lholmes8**
Update so process now uses $KDBHDB environment variable if hdbdir command line parameter is not given.
Add functionality to check there is only one enumeration type across tables in given partitions.
Add email alert if more than one enumeration type is found.

### CODE REVIEW

**Comment by claire-sloan** [comment]
Think this is all meant to be in the one file? 
Add label, assignee and nice description on what functionality has been added/fixed

File: code/checks/hdbcheck.q [hunk 1]
```diff
@@ -5,14 +5,19 @@ usage:@[value;`.hdbcheck.usage;"General:
...
+
+ Optional flags: 
+ [-hdbdir x]: the full path name to the hdb directory (if not supplied $KDBHDB is used)
  [-pars x]: partition name(s) to perform check on (default is to check all partitions) e.g -pars 2018.04.18 2018.04.20"];
 
 if[`help in key .proc.params;-1 usage; exit 0];                                                 / show usage statement if -help flag used
 
-hdbdir:@[value;`hdbdir;first .proc.params.hdbdir];                                              / Use hdbdir if set otherwise take from procs.params
+
+$[`hdbdir in key .proc.params;
```
Comment (ThomasSmyth) [inline code/checks/hdbcheck.q]: This whole bracket can be simplified by pulling out duplicate code:
```
hdbdir:@[value;`hdbdir;]$[`hdbdir in key .proc.params;first .proc.params.hdbdir;getenv`KDBHDB]
```

File: code/checks/hdbcheck.q [hunk 2]
```diff
@@ -5,14 +5,19 @@ usage:@[value;`.hdbcheck.usage;"General:
...
  [-pars x]: partition name(s) to perform check on (default is to check all partitions) e.g -pars 2018.04.18 2018.04.20"];
 
 if[`help in key .proc.params;-1 usage; exit 0];                                                 / show usage statement if -help flag used
 
-hdbdir:@[value;`hdbdir;first .proc.params.hdbdir];                                              / Use hdbdir if set otherwise take from procs.params
+
+$[`hdbdir in key .proc.params; 
+   hdbdir:@[value;`hdbdir;first .proc.params.hdbdir];                                         / Use hdbdir if set otherwise take from procs.param
+   hdbdir:@[value;`hdbdir;getenv[`KDBHDB]]                                                    / Use KDBHDB env var if hdbdir param not set           
+   ]
```
Comment (ThomasSmyth) [inline code/checks/hdbcheck.q]: Needs semi colon.

File: code/checks/hdbcheck.q [hunk 3]
```diff
@@ -41,6 +46,10 @@ update countMatch:doColsMatch each info, symParted:isSymParted each info from `.
 colMismatch:select from .hdbcheck.tabInfo where countMatch=0;
 unpartedSym:select from .hdbcheck.tabInfo where symParted=0;
 
+isSymEnum:{20=type ?[?[$[-11= type y;value y;y];();0b;(enlist x)!enlist x];();();x]};           / Function to test columns for enumeration type
```
Comment (ThomasSmyth) [inline code/checks/hdbcheck.q]: `(enlist x)!enlist x` -> `{x!x}enlist x` or `{x!x}(),x`

File: code/checks/hdbcheck.q [hunk 4]
```diff
@@ -60,4 +69,12 @@ if[count .hdbcheck.unpartedSym; params[`subject]:"no parted attribute on sym col
 if[not count key `:sym; params[`subject]:"no sym file at top level";
   params[`body]:"sym file missing from ",.hdbcheck.hdbdir;
   .email.send params;
- ];                                                                                             / Alert if no sym file on top level
\ No newline at end of file
+  ];                                                                                            / Alert if no sym file on top level
+
+if[sum sum each .hdbcheck.tabEnumTypes; 
+  params[`subject]:"more than one enumeration type";                                            / Alert if more than one enumeration type
+  params[`body]: string @[last partabs[pars];raze where each .hdbcheck.tabEnumTypes];
```
Comment (ThomasSmyth) [inline code/checks/hdbcheck.q]: No need for squares round `pars`.

**Pull request merged**